--- a/src/modules/simulation/gz_bridge/GZBridge.cpp
+++ b/src/modules/simulation/gz_bridge/GZBridge.cpp
@@ -247,9 +247,9 @@
 	// FIMEX: once we're on jetty or later
 	// The magnetometer plugin publishes in units of gauss and in a weird left handed coordinate system
 	// https://github.com/gazebosim/gz-sim/pull/2460
-	report.x = -msg.field_tesla().y();
-	report.y = -msg.field_tesla().x();
-	report.z = msg.field_tesla().z();
+	report.x = msg.field_tesla().x();
+	report.y = -msg.field_tesla().y();
+	report.z = -msg.field_tesla().z();
 
 	_sensor_mag_pub.publish(report);
 }
